go/parquet/internal/utils/bit_block_counter.go

```diff
@@ -0,0 +1,260 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"math"
+	"math/bits"
+	"unsafe"
+
+	"github.com/apache/arrow/go/arrow/bitutil"
+	"github.com/apache/arrow/go/arrow/endian"
+)
+
+var toLEFunc func(uint64) uint64
+
+func init() {
```

Comment:
https://www.digitalocean.com/community/tutorials/understanding-init-in-go

`init` functions are run before any other code in the package is run at import time. In the case here, it would get run once during the global initialization of whatever program imported this.

Reply:
thanks